SWLOR.Game.Server/Feature/AbilityDefinition/Force/BenevolenceAbilityDefinition.cs

```diff
@@ -28,13 +28,27 @@ private void Impact(uint activator, uint target, int baseAmount)
             var targetBonus = willBonus;
             if (target != activator && Stat.GetCurrentFP(activator) >= 16)
             {
-                var willRestore = (willBonus / 2);
+                var willRestore = (willBonus / 2) * 4;
+                var duration = (willBonus * 60f);
+                var effect = EffectRegenerate(willRestore, 6f);
                 Stat.ReduceFP(activator, 10);
                 Stat.ReduceStamina(activator, willRestore);
                 Stat.RestoreFP(target, willRestore);
                 Stat.RestoreStamina(target, willRestore);
-                ApplyEffectToObject(DurationType.Instant, EffectRegenerate(willRestore * 4, 24f), target);
                 targetBonus = willBonus * 4;
+
+
+                for (var e = GetFirstEffect(target); GetIsEffectValid(e); e = GetNextEffect(target))
+                {
+                    if (GetEffectTag(e) == "FORCE_BENEVOLENCE")
```

Comment:
Please assign FORCE_BENEVOLENCE to a constant and reuse that instead of the raw string.